### PR TITLE
Space properties opaque

### PR DESCRIPTION
**Pull request by micbar**
## Description
Change space properties implementation to use extended attributes instead of a spaces.yaml file

needs https://github.com/cs3org/reva/pull/2527 and  https://github.com/cs3org/reva/pull/2512

an needs #3146 to be merged before and rebase

## Description

Properties should be stored by the storage driver and not the graph service.

We can now set and modify short description, space image and space readme. Only managers can set the short description. Editors can change the space image and readme id.

## List spaces
```sh
curl -L -X GET 'https://localhost:9200/graph/v1.0/me/drives/' \
-H 'Authorization: Basic YWRtaW46YWRtaW4='
```
### Response
```sh
{
    "value": [
        {
            "driveType": "public",
            "id": "7993447f-687f-490d-875c-ac95e89a62a4",
            "lastModifiedDateTime": "1970-01-01T01:00:00+01:00",
            "name": "Public shares",
            "root": {
                "id": "7993447f-687f-490d-875c-ac95e89a62a4",
                "webDavUrl": "https://localhost:9200/dav/spaces/7993447f-687f-490d-875c-ac95e89a62a4"
            }
        },
        {
            "description": "Admins Space",
            "driveType": "personal",
            "id": "ddc2004c-0977-11eb-9d3f-a793888cd0f8",
            "lastModifiedDateTime": "2022-02-10T13:50:32.203617+01:00",
            "name": "Test6",
            "owner": {
                "user": {
                    "id": "ddc2004c-0977-11eb-9d3f-a793888cd0f8"
                }
            },
            "quota": {
                "remaining": 70769467392,
                "state": "normal",
                "total": 70771960968,
                "used": 2493576
            },
            "root": {
                "id": "ddc2004c-0977-11eb-9d3f-a793888cd0f8",
                "permissions": [
                    {
                        "grantedTo": [
                            {
                                "user": {
                                    "id": "ddc2004c-0977-11eb-9d3f-a793888cd0f8"
                                }
                            }
                        ],
                        "roles": [
                            "manager"
                        ]
                    }
                ],
                "webDavUrl": "https://localhost:9200/dav/spaces/ddc2004c-0977-11eb-9d3f-a793888cd0f8"
            },
            "special": [
                {
                    "eTag": "\"ce0e9fa86a616a04e3426b425b7f7ecd\"",
                    "file": {
                        "mimeType": "text/markdown"
                    },
                    "id": "b1278e81-fc7f-40f9-ae21-0c823a9edc76",
                    "lastModifiedDateTime": "2022-02-04T20:36:52.000000738+01:00",
                    "name": "README.md",
                    "size": 6783,
                    "specialFolder": {
                        "name": "readme"
                    },
                    "webDavUrl": "https://localhost:9200/dav/spaces/ddc2004c-0977-11eb-9d3f-a793888cd0f8/README.md"
                },
                {
                    "eTag": "\"1b414efd23ee31ea29ce1d2d208a9991\"",
                    "file": {
                        "mimeType": "image/png"
                    },
                    "id": "c778e53f-1ed0-4962-af55-98f9f1162f99",
                    "lastModifiedDateTime": "2021-12-02T11:47:15.000000603+01:00",
                    "name": "1638442035562.png",
                    "size": 465337,
                    "specialFolder": {
                        "name": "image"
                    },
                    "webDavUrl": "https://localhost:9200/dav/spaces/ddc2004c-0977-11eb-9d3f-a793888cd0f8/1638442035562.png"
                }
            ]
        },
        {
            "driveType": "virtual",
            "id": "a0ca6a90-a365-4782-871e-d44447bbc668",
            "name": "Shares Jail",
            "quota": {
                "remaining": 0,
                "state": "exceeded",
                "total": 0,
                "used": 0
            },
            "root": {
                "id": "a0ca6a90-a365-4782-871e-d44447bbc668",
                "webDavUrl": "https://localhost:9200/dav/spaces/a0ca6a90-a365-4782-871e-d44447bbc668"
            }
        }
    ]
}
```
## Update a Space

```sh
curl -L -X PATCH 'https://localhost:9200/graph/v1.0/drives/ddc2004c-0977-11eb-9d3f-a793888cd0f8' \
-H 'Authorization: Basic YWRtaW46YWRtaW4=' \
-H 'Content-Type: application/json' \
--data-raw '{
  "Name": "Test6",
  "Description": "foo",
  "special":[
      {
          "specialFolder": {
              "name": "image"
          },
          "id": "ffc2004c-0977-11eb-9d3f-a793888cd0f8"
      },
      {
          "specialFolder": {
              "name": "readme"
          },
          "id": "fd42004c-0977-11eb-9d3f-a793888cd0f8"
      }
  ]
}'
````

## How Has This Been Tested?
- [x] manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/9305/20/1) failed. Further test are cancelled...

File: graph/pkg/service/v0/drives.go [hunk 1]
```diff
@@ -270,6 +275,7 @@ func (g Graph) UpdateDrive(w http.ResponseWriter, r *http.Request) {
 		},
 	}
 
+	// Use the Opaque prop of the request
```
[the remark this answers is not in the document]
Comment (micbar) [reply graph/pkg/service/v0/drives.go:278]: maybe not clear enough, here we are using the *request*

File: graph/pkg/service/v0/drives.go [hunk 2]
```diff
@@ -281,6 +287,25 @@ func (g Graph) UpdateDrive(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Use the Opaque prop of the space
```
[the remark this answers is not in the document]
Comment (micbar) [reply graph/pkg/service/v0/drives.go:290]: this uses a different opaque than above, it uses the space itself. It is easy to get that wrong.

File: graph/pkg/service/v0/drives.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (C0rby) [reply graph/pkg/service/v0/drives.go:278]: Ah, now I see that you want to communicate that you are using a different opaque than below.

[the remark this answers is not in the document]
Comment (micbar) [reply graph/pkg/service/v0/drives.go]: now a slightly different wording.

File: graph/pkg/service/v0/drives.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (micbar) [reply graph/pkg/service/v0/drives.go]: same as above

File: graph/pkg/service/v0/driveitems.go [hunk 3]
```diff
@@ -142,3 +153,48 @@ func cs3ResourceToDriveItem(res *storageprovider.ResourceInfo) (*libregraph.Driv
...
+			readmeItem, err := g.getDriveItem(ctx, &storageprovider.ResourceId{StorageId: space.Root.StorageId, OpaqueId: readmeID})
+			if err != nil {
+				g.logger.Error().Err(err).Str("ID", readmeID).Msg("Could not get readme Item")
+			} else {
+				readmePath, err := g.getPathForDriveItem(ctx, &storageprovider.ResourceId{StorageId: space.Root.StorageId, OpaqueId: readmeID})
+				if err != nil {
+					g.logger.Error().Err(err).Str("ID", readmeID).Msg("Could not get readme path")
+				} else {
+					readmeItem.SpecialFolder = &libregraph.SpecialFolder{Name: libregraph.PtrString(ReadmeSpecialFolderName)}
+					readmeItem.WebDavUrl = libregraph.PtrString(baseURL.String() + filepath.Join(space.Root.OpaqueId, *readmePath))
```
[the remark this answers is not in the document]
Comment (micbar) [reply graph/pkg/service/v0/driveitems.go]: changed

**Review by C0rby — APPROVED** [review]
LGTM

**Comment by JanAckermann** [comment]
nice :+1:

**Pull request merged**